### ISSUE TITLE
Move Google Forms from Google Drive to Google Drive

### ISSUE THREAD

**Issue by ahmedadelfarag** [rclone/rclone#6302]
`rclone move Source:\source dest:\dest --log-file rclone/logs/test.log --drive-server-side-across-configs -vv -P`

I found that all Google forms files are not moved and it gives me the following error.

`No export formats found for "application/vnd.google-apps.form"`

Check this : https://forum.rclone.org/t/move-files-on-from-google-drive-to-google-drive/31722

**Comment by timwsuqld** [comment]
I'm hitting this same issue. We may need a filter for files we can and can't move? e.g. can we move shortcuts?

**Comment by ncw** [comment]
A possibility would be to add a flag to consider all Google docs whether they can be downloaded or not.

This would enable a server side copy or move of a google form to succeed.

**Comment by abhiraj2** [comment]
I would like to give this a try, if its alright @ncw

**Comment by ncw** [comment]
@abhiraj2 that would be great. Let me know if you want any hints.

**Comment by abhiraj2** [comment]
sure hints would be great, cause this looks a lil trickier than what I have contributed before 😅

**Comment by ncw** [comment]
If you look through the drive backend code you'll see that it ignores objects without an export format.

Try disabling that check and see if that's enough to get server side copies of forms working then report back!

**Comment by abhiraj2** [comment]
sure thing

**Comment by ferferga** [comment]
@ncw With disabling the check, you mean in the code or there's already a way to do it through config or ``backend`` commands and I'm missing out?

**Comment by ncw** [comment]
Give this a try --drive-show-all-gdocs - that should allow you to server side copy forms.

[v1.65.0-beta.7393.95a6101a5.fix-6302-drive-gdocs](https://beta.rclone.org/branch/fix-6302-drive-gdocs/v1.65.0-beta.7393.95a6101a5.fix-6302-drive-gdocs/) on branch [fix-6302-drive-gdocs](https://github.com/rclone/rclone/tree/fix-6302-drive-gdocs) (uploaded in 15-30 mins)

#### --drive-show-all-gdocs

Show all Google Docs including non-exportable ones in listings.

If you try a server side copy on a Google Form without this flag, you
will get this error:

    No export formats found for "application/vnd.google-apps.form"

However adding this flag will allow the form to be server side copied.

Note that rclone doesn't add extensions to the Google Docs file names
in this mode.

Do **not** use this flag when trying to download Google Docs - rclone
will fail to download them.


Properties:

- Config:      show_all_gdocs
- Env Var:     RCLONE_DRIVE_SHOW_ALL_GDOCS
- Type:        bool
- Default:     false

**Comment by ferferga** [comment]
@ncw It doesn't work perfectly. This is the command I'm running

```
rclone copy Source: TeamDriveDest: --progress --server-side-across-configs --drive-show-all-gdocs --drive-skip-dangling-shortcuts
```

**Details**:
- *Source* is a normal Drive and *TeamDriveDest* a team drive the user of *source* is joined to.
- Credentials are shared: I configured *Source* through `rclone config`, copied it to *TeamDriveDest* and edited *TeamDriveDest* config to be a shared drive.

**The caveats I'm experiencing after a **:
- *Maybe it's not relevant to this issue, but pointing it out just in case, since this makes the command exit with errors, which might be relevant:* Although I'm using the `--drive-skip-dangling-shortcuts` flag, I'm getting printed to the console:
```
2023-09-29 09:46:38 NOTICE: Dangling shortcut *X* detected
```
- After the command finishes, in *Source* I'm seeing that rclone leaves the copied forms (named as Google Drive's interface does it: ``Copy of *FormName*``)
- There are both the ``Copy of *FormName*`` and `*FormName*` files in the destination.

**Comment by ncw** [comment]
> * _Maybe it's not relevant to this issue, but pointing it out just in case, since this makes the command exit with errors, which might be relevant:_ Although I'm using the `--drive-skip-dangling-shortcuts` flag, I'm getting printed to the console:
> 
> ```
> 2023-09-29 09:46:38 NOTICE: Dangling shortcut *X* detected
> ```

Yes you'll still get this warning even with  `--drive-skip-dangling-shortcuts`. It doesn't affect anything though or make the command exit with errors.

> * After the command finishes, in _Source_ I'm seeing that rclone leaves the copied forms (named as Google Drive's interface does it: `Copy of *FormName*`)
> 
> * There are both the `Copy of *FormName*` and `*FormName*` files in the destination.

That is very strange! Rclone doesn't have any logic to make "Copy of FormName" files in it so that must be google drive doing it.

I tried to replicate this with a single form on a mydrive and server side copying it to a shared drive, but it worked fine.

Can you try to replicate just with a single form in a single directory?

**Comment by ferferga** [comment]
@ncw You're right, it works flawlessly in that case!

I need to investigate what exactly causes this behaviour, since I think I saw it in other files too. To be honest, what I'm trying to copy is a rather complex setup: they're shortcuts pointing to shared folders/files, so probably there's something else in the equation (as there's a lot going on in my setup).

Let's see if I can find a common denominator to that issue, then I'll open a new one. In the meantime, I think that branch is safe to merge! (and a much needed addition)

Thank you very much!

**Comment by ncw** [comment]
I've merged this to master now which means it will be in [the latest beta](https://beta.rclone.org/) in 15-30 minutes and released in v1.65

If you can figure out something else which needs fixing in rclone then please open another issue :-)
